src/life.rs

```diff
@@ -891,16 +892,26 @@ impl WorldState {
 
 }
 
+fn next_color(color: Color) -> Color {
+    match color {
+        Color::Yellow => Color::Cyan,
+        Color::Cyan => Color::Yellow,
+        _ => unreachable!()
+    }
+}
```

Comment:
I think there a [Cycle iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cycle) that we might want for this sort of thing?

```
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ cat cycle_experiment.rs 
fn main() {
    let cycle_items = ["a", "b", "c"];
    for (index, item) in (0..10).zip(cycle_items.iter().cycle()) {
        println!("{}: {}", index, item);
    }
}
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ rustc cycle_experiment.rs 
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ ./cycle_experiment 
0: a
1: b
2: c
3: a
4: b
5: c
6: a
7: b
8: c
9: a
```